src/IssueViz/TableControls/VSTableEventProcessorProvider.cs

```diff
@@ -0,0 +1,142 @@
+﻿/*
+ * SonarQube Client
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using System.Linq;
+using Microsoft.VisualStudio.Shell.TableControl;
+using Microsoft.VisualStudio.Shell.TableManager;
+using Microsoft.VisualStudio.Text.Classification;
+using Microsoft.VisualStudio.Utilities;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.Analysis;
+
+namespace SonarLint.VisualStudio.IssueVisualization.TableControls
+{
+    [Export(typeof(ITableControlEventProcessorProvider))]
+    [Name(nameof(VSTableEventProcessorProvider))]
+    [Order(Before = Priority.Default)]
+    // We can specify multiple data source types and data source i.e. the same provider can be used for
+    // multiple table controls/sources
+    [DataSourceType(StandardTableDataSources.ErrorTableDataSource)]
```

Comment:
Do we have to specify DataSourceType? (e.g. would it work / make a difference in perf with just DataSource?)
If not, then will be a bit "cleaner" not to specify the type, so we could just say that we support source A and source B, without necessarily knowing their type.

Reply:
I think "type" in this context refers to the table control to which the data source is attached.
The attribute is required. If we want to specify that we are interested in every table control then we'd specify `StandardTableDataSources.AnyDataSource`.